apps/main/src/lend/hooks/useBorrowPositionDetails.ts

```diff
@@ -0,0 +1,131 @@
+import meanBy from 'lodash/meanBy'
+import { useMemo } from 'react'
+import { useMarketOnChainRates } from '@/lend/entities/market-details'
+import { useUserLoanDetails } from '@/lend/entities/user-loan-details'
+import networks from '@/lend/networks'
+import useStore from '@/lend/store/useStore'
+import { ChainId, OneWayMarketTemplate } from '@/lend/types/lend.types'
+import type { Address, Chain } from '@curvefi/prices-api'
+import { useLendingSnapshots } from '@ui-kit/entities/lending-snapshots'
+import { useWallet } from '@ui-kit/features/connect-wallet'
+import { BorrowPositionDetailsProps } from '@ui-kit/features/market-position-details/BorrowPositionDetails'
+import { useTokenUsdRate } from '@ui-kit/lib/model/entities/token-usd-rate'
+
+type UseBorrowPositionDetailsProps = {
+  chainId: ChainId
+  market: OneWayMarketTemplate | null | undefined
+  marketId: string
+}
+
+export const useBorrowPositionDetails = ({
+  chainId,
+  market,
+  marketId,
+}: UseBorrowPositionDetailsProps): BorrowPositionDetailsProps => {
+  const { wallet } = useWallet()
+  const { data: userLoanDetails, isLoading: isUserLoanDetailsLoading } = useUserLoanDetails({
+    chainId,
+    marketId,
+    userAddress: wallet?.account.address,
+  })
+  const marketRate = useStore((state) => state.markets.ratesMapper[chainId]?.[marketId])
+  const prices = useStore((state) => state.markets.pricesMapper[chainId]?.[marketId])
+
+  const { data: onChainRatesData, isLoading: isOnchainRatesLoading } = useMarketOnChainRates({
+    chainId: chainId,
+    marketId,
+  })
+  const { data: collateralUsdRate, isLoading: collateralUsdRateLoading } = useTokenUsdRate({
+    chainId: chainId,
+    tokenAddress: market?.addresses?.collateral_token,
+  })
+  const { data: borrowedUsdRate, isLoading: borrowedUsdRateLoading } = useTokenUsdRate({
+    chainId: chainId,
+    tokenAddress: market?.addresses?.borrowed_token,
+  })
+
+  const { data: lendSnapshots, isLoading: isLendSnapshotsLoading } = useLendingSnapshots({
+    blockchainId: networks[chainId].id as Chain,
+    contractAddress: market?.addresses?.controller as Address,
+  })
+
+  const thirtyDayAvgRate = useMemo(() => {
+    if (!lendSnapshots) return null
+
+    const thirtyDaysAgo = new Date()
+    thirtyDaysAgo.setDate(thirtyDaysAgo.getDate() - 30)
+
+    const recentSnapshots = lendSnapshots.filter((snapshot) => new Date(snapshot.timestamp) > thirtyDaysAgo)
+
+    if (recentSnapshots.length === 0) return null
+
+    return meanBy(recentSnapshots, ({ borrowApy }) => borrowApy) * 100
+  }, [lendSnapshots])
+
+  const borrowApy = onChainRatesData?.rates?.borrowApy ?? marketRate?.rates?.borrowApy
+
+  const collateralTotalValue = useMemo(() => {
+    if (!collateralUsdRate || !userLoanDetails?.state?.collateral) return null
```

Comment:
There's a lot of `userLoanDetails?.state?.yadda`. Maybe you can destructure the properties you need a the top level so the code can get a lot shorter and avoid repetition?
Stuff like:

`const { collateral, borrowed, pnl } = userLoanDetails?.state ?? {}`

This holds for all hooks in this folder